File: JavaLox/src/com/craftinginterpreters/lox/ExternalLoxGlobal.java
package com.craftinginterpreters.lox;

import java.util.HashMap;
import java.util.Map;

/**
 *
 * @author brusc
 */
class ExternalLoxGlobal {
        public Map<String, Object> externalGlobals;

        public ExternalLoxGlobal()
        {
            externalGlobals = new HashMap<>();
        }

        public ExternalLoxGlobal(HashMap<String, Object> externalGlobals)
        {
            this.externalGlobals = externalGlobals;
        }

        public void defineGlobals(Environment loxGlobals)
        {
            if (externalGlobals != null)
            {
                for (String key : externalGlobals.keySet())
                {
                    loxGlobals.define(key, externalGlobals.get(key));
                }
            }
        }
}

File: CsLox/com/craftinginterpreters/lox/ExternalLoxGlobal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.craftinginterpreters.lox
{
    internal class ExternalLoxGlobal : HandleLoxGlobals
    {
        public Dictionary<String, Object> externalGlobals;

        public ExternalLoxGlobal()
        {
            externalGlobals = new Dictionary<String, Object>();
        }

        public ExternalLoxGlobal(Dictionary<String, Object> externalGlobals)
        {
            this.externalGlobals = externalGlobals;
        }

        public void defineGlobals(Environment loxGlobals)
        {
            if (externalGlobals != null)
            {
                foreach (String key in externalGlobals.Keys)
                {
                    loxGlobals.define(key, externalGlobals[key]);
                }
            }
        }
    }
}
